base/libdl.jl

```diff
@@ -205,6 +205,43 @@ if is_linux()
     end
 end # linux-only
 
+if Sys.KERNEL == :FreeBSD
```

Comment:
Is this entirely unique to FreeBSD, or will it apply to Dragonfly and other BSDs? If the latter, this should be `if is_bsd() && !is_apple()`.